tests/test_integration/test_wf_create_case_metadata.py

```diff
@@ -1,21 +1,32 @@
-import subprocess
+import getpass
+import os
+import sys
 
+import ert.__main__
+import pytest
 import yaml
 
 
-def test_create_case_metadata_runs_successfully(fmu_snakeoil_project, monkeypatch):
-    monkeypatch.chdir(fmu_snakeoil_project / "ert/model")
-    with open("snakeoil.ert", "a", encoding="utf-8") as f:
+def _add_create_case_workflow(filepath):
+    with open(filepath, "a", encoding="utf-8") as f:
         f.writelines(
             [
                 "LOAD_WORKFLOW ../bin/workflows/xhook_create_case_metadata\n"
                 "HOOK_WORKFLOW xhook_create_case_metadata PRE_SIMULATION\n"
             ]
         )
-    run_result = subprocess.run(
-        ["ert", "test_run", "snakeoil.ert", "--disable-monitoring"],
+
+
+def test_create_case_metadata_runs_successfully(
+    fmu_snakeoil_project, monkeypatch, mocker
+):
+    monkeypatch.chdir(fmu_snakeoil_project / "ert/model")
+    _add_create_case_workflow("snakeoil.ert")
+
+    mocker.patch(
+        "sys.argv", ["ert", "test_run", "snakeoil.ert", "--disable-monitoring"]
     )
-    assert run_result.returncode == 0
+    ert.__main__.main()
```

Comment:
Will at least cache the import so we're not running a new child process each time, but maybe a small timeframe over an internal package dependency -- though unlikely to change